Improve alias configutation APIs (#11468)
pydantic/pydantic: docs/errors/usage_errors.md

except PydanticUserError as exc_info:
    assert exc_info.code == 'invalid-self-type'
```

> ## `validate_by_alias` and `validate_by_name` `False` {#validate-by-alias-and-name-false}

Reviewer: ```suggestion
## `validate_by_alias` and `validate_by_name` both set to `False` {#validate-by-alias-and-name-false}
```

> This error is raised when you set `validate_by_alias` and `validate_by_name` to `False` in a model's configuration.

Reviewer: ```suggestion
This error is raised when you set `validate_by_alias` and `validate_by_name` to `False` in the configuration
```

As configuration is not a concept limited to models.
